vignettes/plot_projection.Rmd

```diff
@@ -0,0 +1,283 @@
+
+The following vignette describes the principal usage of the 
+`plot_step_ahead_forecasts()` function.
+
+```{r lib}
+library(hubVis)
+```
+
+Plots are available in two output formats:
+
+- "interactive" format: a [Plotly](https://plotly.com) output object with interactive legend, hover
+text, zoom-in and zoom-out options, etc.
+- "static" format: a [ggplot2](https://ggplot2.tidyverse.org/) output object.
+By default, the output plot is "interactive", but it can be changed to "static"
+by setting  the `interactive` parameter to FALSE. See end of the document for examples.
+
+## Load and Filter Data
+
+The package contains 2 datasets that can be used for examples:
+
+- `example_round1.csv`: example of model output for a round associated with the 
+origin date: "2021-03-07" (called "round 1"), target: "incident case", for the
+US national level, from the
+[example-complex-scenario-hub](https://github.com/Infectious-Disease-Modeling-Hubs/example-complex-scenario-hub). The data set also contains an ensemble calculated by applying the 
+function: `hubEnsembles::simple_ensemble(df_round1, agg_fun = "median")`
+
+- `truth_data.csv`: example of target data from the [example-complex-scenario-hub](https://github.com/Infectious-Disease-Modeling-Hubs/example-complex-scenario-hub). The data here comes from the `"target-data/US_inc_case.csv"` file.
+
+### Load data
+
+```{r load_data}
+projection_path <- system.file("example_round1.csv", package = "hubVis")
+projection_data <- read.csv(projection_path)
+head(projection_data)
+
+truth_path <- system.file("truth_data.csv", package = "hubVis")
+truth_data <- read.csv(truth_path)
+head(truth_data)
+```
+
+### Data Preparation
+
+To fit the input format of the function, a `target_date` column needs to be 
+calculated for the model projections data.
+
+```{r data_prep}
+projection_data <- dplyr::mutate(
+  projection_data, target_date = as.Date(origin_date) + (horizon * 7) - 1)
+head(projection_data)
+```
+
+## Plot
+
+The function requires only 2 parameters:
+
+- `forecast_data`: a `model_output_df` object containing all the Hubverse
+ standard columns, including a `"target_date"`, and a `"model_id"` column in the 
+`hub_mdl_out_df format`. As all forecasts in forecast_data will be plotted, 
+ all filtering needs to happen outside this function.
+- ` truth_data`: a `data.frame` object containing the ground truth data, including 
+the columns: `time_idx` and `value`.
+
+### "Simple" plot
+
+The `"projection_data"` and `"truth_data"` contain information for multiple 
+locations, and scenarios.
+
+To plot the model projections for the US, Scenario A:
+
+```{r A_us}
+# Pre-filtering
+projection_data_A_us <- dplyr::filter(projection_data, 
+                                      scenario_id == "A-2021-03-05", 
+                                      location == "US")
+# Class update
+projection_data_A_us <- hubUtils::as_model_out_tbl(projection_data_A_us)
```

Comment:
Noting that when I run this line of code, I get an error
```
Error in `[.tbl_df`(tbl, , c(std_colnames[1], names(tbl)[!names(tbl) %in%  : 
  object 'std_colnames' not found
```
That I think has to do with hubUtils needing to be loaded so it can access `std_colnames`. I will add this above with a suggested edit/PR.

Reply:
I think there is an open issue on hubUtils, [issue 96.](https://github.com/Infectious-Disease-Modeling-Hubs/hubUtils/issues/96). I thought it was solved in an recent version of the package. A solution will be to either update the DESCRIPTION file here to force a latest version of hubUtils or to update the vignette documentation. I can run some tests to identify which version is required

Reply:
Since it doesn't seem like this has been solved yet, I'm just going to load the `hubUtils` package in the vignette so it builds.